types/lib/client.d.ts

```diff
@@ -0,0 +1,190 @@
+/// <reference types="node" />
+/**
+ * Module dependencies
+ */
+import * as events from 'events'
+import {
+  IClientOptions,
+  IClientPublishOptions,
+  IClientSubscribeOptions
+} from './client-options'
+import { Store } from './store'
+import { Packet, QoS } from './types'
+export interface ISubscriptionGrant {
+  /**
+   *  is a subscribed to topic
+   */
+  topic: string
+  /**
+   *  is the granted qos level on it, may return 128 on error
+   */
+  qos: QoS | number
+}
+export interface ISubscriptionRequest {
+  /**
+   *  is a subscribed to topic
+   */
+  topic: string
+  /**
+   *  is the granted qos level on it
+   */
+  qos: QoS
+}
+export interface ISubscriptionMap {
+  /**
+   * object which has topic names as object keys and as value the QoS, like {'test1': 0, 'test2': 1}.
+   */
+  [topic: string]: QoS
+}
+
+export declare type ClientSubscribeCallback = (err: Error, granted: ISubscriptionGrant[]) => void
+export declare type OnMessageCallback = (topic: string, payload: Buffer, packet: Packet) => void
+export declare type OnPacketCallback = (packet: Packet) => void
+export declare type OnErrorCallback = (error: Error) => void
+export declare type PacketCallback = (error?: Error, packet?: Packet) => any
+
+// export interface IReinterval {
+//   clear: () => void
+//   reschedule (ms: number): any
+// }
+export interface IStream extends events.EventEmitter {
+  pipe (to: any): any
+  destroy (): any
+  end (): any
+}
+/**
+ * MqttClient constructor
+ *
+ * @param {Stream} stream - stream
+ * @param {Object} [options] - connection options
+ * (see Connection#connect)
+ */
+export declare class MqttClient extends events.EventEmitter {
+  // public connackTimer: NodeJS.Timer
```

Comment:
these weren't prefixed with a `_` so I marked them public (default in TypeScript if not specifide) (the first tslint configuration, before I used the `standard` variant wanted explicit so I just used the --fix options) 

In any case, I guess we don't really want them exposed?